client/main.go

```diff
@@ -106,15 +107,16 @@ func handleUserInput(rw *bufio.ReadWriter, handlers map[string]UserHandleFunc) {
 	if len(cmds) < 2 || handlerFunc == nil {
 		log.Println("invalid input: " + input)
 		log.Println(cmds[0])
-	} else {
-		handlerFunc(rw.Writer, cmds[1])
+		return
 	}
+
+	handlerFunc(rw.Writer, cmds[1])
 }
 
-func handleGetRooms(reader *bufio.Reader, state *State) {
+func handleGetRooms(reader *bufio.Reader, _ *State) {
 	rooms := make([]string, 10)
```

Comment:
Usually don't need to give a size when making slices, especially when passing to be decoding into. The underlying array of the slice might just be reallocated as needed (I might be wrong). See https://blog.golang.org/go-slices-usage-and-internals for details about slices, if you're interested.

Reply:
If no size is needed, `var rooms []string` can be used which creates a nil slice. I have no idea why it works, though, because in the next line I refer to the address of `rooms`. As I understand things, a nil slice doesn't have an address.